batch/create-batch-tables.sql

```diff
@@ -359,6 +359,9 @@ BEGIN
             (SELECT SUM(cores_mcpu) FROM jobs
              WHERE jobs.state = 'Ready' AND jobs.batch_id = in_batch_id),
             0);
+      UPDATE jobs
+        SET jobs.state = IF(jobs.n_pending_parents = 0, 'Ready', 'Pending')
+        WHERE jobs.batch_id = in_batch_id;
```

Comment:
So I'm worried about this.  What if there are 10M jobs?  In the design I was trying to avoid bulk operations like this.  One option is to have another index/table that records ready jobs for open batches, then have a loop like the cancel and schedule loops that grabs limited group of jobs that are ready from closed batches (e.g. 100) and sets them to ready.